fabric-armor-api-v1/src/main/java/net/fabricmc/fabric/api/client/armor/v1/CustomTexturedArmor.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.armor.v1;
+
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Implement this interface on an Item which you would like to have a custom textured armor model.
+ */
+public interface CustomTexturedArmor {
```

Comment:
Honestly I think just implementing the interface and using the item's registry namespace and path would probably suffice over a whole method.

Reply:
Well possibly, I may default the return value to the item registry namespace? Since I really want the customizability here without limiting due to the item registry namespace.

It will be useful for items that use the same texture (and maybe using the different UV).

Reply:
I could see a reason making the selected texture stack aware and slightly more contextual.